File: SimpleChaos/demo-applications/spring-demo/src/main/java/com/example/demo/RestEndpoint.java
package com.example.demo;

import com.languagelatte.simplechaos.spring.ChaosService;

import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.ResponseBody;
import org.springframework.web.bind.annotation.RestController;

@Controller
@RestController
@RequestMapping("/simplechaos")
public class RestEndpoint {

    private final ChaosService chaosService;

    public RestEndpoint(ChaosService chaosService){
        this.chaosService = chaosService;
    }
    
    @RequestMapping(value = "/exception", method = RequestMethod.GET)
    @ResponseBody
    public ResponseEntity<String> exception() {  

        String response;
        try {   
            chaosService.exception();
            response = "No Exception";
        }catch (Exception e){
               response = e.getMessage();
        }
        return new ResponseEntity<String>(response , HttpStatus.OK);
    }

    @RequestMapping(value = "/error", method = RequestMethod.GET)
    @ResponseBody
    public ResponseEntity<String> error() {  

        String response;
        try {   
            chaosService.error();
            response = "No Error";
        }catch (Error e){
               response = e.getMessage();
        }
        return new ResponseEntity<String>(response , HttpStatus.OK);
    }

    @RequestMapping(value = "/latency", method = RequestMethod.GET)
    @ResponseBody
    public ResponseEntity<String> latency() {  

        String response;
        
        long start = System.currentTimeMillis();
        chaosService.latency();
        response = "Time Taken = " + (System.currentTimeMillis() - start);
        
        return new ResponseEntity<String>(response , HttpStatus.OK);
    }

    @RequestMapping(value = "/jvmcrash", method = RequestMethod.GET)
    @ResponseBody
    public ResponseEntity<String> jvmcrash() {  

        String response = "No Jvm Shutdown";
        
        chaosService.jvmCrash();
         
        return new ResponseEntity<String>(response , HttpStatus.OK);
    }
    
}

File: SimpleChaos/demo-applications/kotlin-spring-demo/src/main/kotlin/com/example/demo/RestEndpoint.kt
package com.example.demo

import com.languagelatte.simplechaos.spring.ChaosService
import org.springframework.web.bind.annotation.RestController
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestMethod
import org.springframework.web.bind.annotation.ResponseBody
import org.springframework.http.ResponseEntity
import org.springframework.http.HttpStatus
import org.springframework.stereotype.Controller
import java.lang.Error
import java.lang.Exception

@Controller
@RestController
@RequestMapping("/simplechaos")
class RestEndpoint(private val chaosService: ChaosService) {
    @RequestMapping(value = ["/exception"], method = [RequestMethod.GET])
    @ResponseBody
    fun exception(): ResponseEntity<String?> {
        var response: String
        try {
            chaosService.exception()
            response = "No Exception"
        } catch (e: Exception) {
            response = if (e.message != null) {
                e.message!!
            } else {
                "exception"
            }

        }
        return ResponseEntity(response, HttpStatus.OK)
    }

    @RequestMapping(value = ["/error"], method = [RequestMethod.GET])
    @ResponseBody
    fun error(): ResponseEntity<String?> {
        var response: String
        try {
            chaosService.error()
            response = "No Error"
        } catch (e: Error) {
            response = if (e.message != null) {
                e.message!!
            } else {
                "error"
            }
        }
        return ResponseEntity(response, HttpStatus.OK)
    }

    @RequestMapping(value = ["/latency"], method = [RequestMethod.GET])
    @ResponseBody
    fun latency(): ResponseEntity<String> {
        val response: String
        val start = System.currentTimeMillis()
        chaosService.latency()
        response = "Time Taken = " + (System.currentTimeMillis() - start)
        return ResponseEntity(response, HttpStatus.OK)
    }

    @RequestMapping(value = ["/jvmcrash"], method = [RequestMethod.GET])
    @ResponseBody
    fun jvmcrash(): ResponseEntity<String> {
        val response = "No Jvm Shutdown"
        chaosService.jvmCrash()
        return ResponseEntity(response, HttpStatus.OK)
    }
}